language/diem-tools/writeset-transaction-generator/src/release.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use diem_types::{
+    access_path::AccessPath,
+    transaction::{ChangeSet, WriteSetPayload},
+    write_set::{WriteOp, WriteSetMut},
+};
+use std::collections::{BTreeMap, BTreeSet};
+use vm::CompiledModule;
+
+pub fn create_release_writeset(
```

Comment:
Would it make sense to add this as a function of `ChangeSet`? (e.g., `ChangeSet::from_modules(old: Vec<CompiledModule>, new: Vec<CompiledModule>`)? This seems like it might be useful in other contexts too

Reply:
My question is if `diem_types` should import `vm::CompiledModule` as a dependency. Thanks to our previous separation `vm` no longer depends on `diem_types` so it is doable. Although I would vote against it to keep the dependency of `diem_types` to be minimal as it is the core dependency for most of our codebase.

Reply:
Fair point. If we find other uses for this, we can find somewhere more general to put it.